allennlp/data/dataset_readers/ccgbank.py

```diff
@@ -0,0 +1,124 @@
+from typing import Dict, List
+import logging
+import re
+
+from overrides import overrides
+
+from allennlp.common.file_utils import cached_path
+from allennlp.common.params import Params
+from allennlp.data.dataset_readers.dataset_reader import DatasetReader
+from allennlp.data.fields import Field, TextField, SequenceLabelField
+from allennlp.data.instance import Instance
+from allennlp.data.token_indexers import TokenIndexer, SingleIdTokenIndexer
+from allennlp.data.tokenizers import Token
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DatasetReader.register("ccgbank")
+class CcgBankDatasetReader(DatasetReader):
+    """
+    Reads data from the CCGbank dataset. In particular, reads the flattened "leaves"
```

Comment:
Not too sure what `reads the flattened "leaves"` means --- are you referring to the fact that the input is a linearized tree?
Also: perhaps we should mention that it reads data from a file where each line is a tree from the ccgbank dataset or something? Since this doesn't read straight from ccgbank.